sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -313,7 +313,7 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (o == null || !(o instanceof Row)) {
```

Comment:
nit:
null is never an instanceof Row already so you should be able to do
```
if (!(o instanceof Row)) {
  return false;
}
```
and drop the null check.

Reply:
Done